builtin/providers/vault/provider.go

```diff
@@ -90,6 +90,7 @@ func Provider() terraform.ResourceProvider {
 			"vault_auth_backend":   authBackendResource(),
 			"vault_generic_secret": genericSecretResource(),
 			"vault_policy":         policyResource(),
+			"vault_mount":          mountResource(),
```

Comment:
This is a good thought, @bmoylan! I think though that since the term "mount" is what appears in Vault's UI it makes sense to follow that here so that it's easier to understand how the Terraform resources map onto the Vault CLI operations.